experimental/addons/gitops/definitions/gitops.cue

```diff
@@ -0,0 +1,46 @@
+gitops: {
+	annotations: {}
+	attributes: workload: definition: {
+		apiVersion: "apps/v1"
+		kind: "Deployment"
+	}
+	description: "KubeVela addon for implementing GitOps for continuous deployment using either fluxcd or argocd"
+	type: "component"
+	labels: {}
+}
+
+template: {
+	output: {
+		if parameter.agent == "fluxcd" {
+			apiVersion: "source.toolkit.fluxcd.io/v1beta2"
+		}
+		if parameter.agent == "argocd" {
+			apiVersion: "argoproj.io/v1alpha1"
+		}
+		kind: ""
+		metadata: {
+			name: context.name
+			namespace: context.namespace
+		}
+		spec: {
+			interval: parameter.pullInterval
+			sourceRef: {
+				if parameter.repoType == "git" {
+					kind: "GitRepository"
+				}
+				if parameter.repoType == "oss" {
+					kind: "Bucket"
+				}
+				name: context.name
+				namespace: context.namespace
+			}
+			path: parameter.paths.glob
+			suspend: parameter.suspend
+			prune: parameter.prune
```

Comment:
Are these fields the same in the argocd and fluxcd?

Reply:
Yes, most of them are. I modified the path though. Fluxcd has just `parameter.paths`, but I added `parameter.paths.glob` so that the user can specify a generic file or folder pattern for system to match. I can change it though.